### PR TITLE
Fixing value mapping for blob and year in Reader

### PR DESCRIPTION
**Pull request by VardhanThigle**
1. The value mapping for blob and year in the reader was not mapping the output to the expected avro type (`string` instead of `char[]` and `integer` instead of `long`), leading to a crash. Fixing the same.
Along with the UT, this PR has been tested on a live db with these two types.
2. With the code review progression, also added UT to cover null values for all types.

### CODE REVIEW

File: v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapperTest.java [hunk 1]
```diff
@@ -240,7 +240,7 @@ private static ImmutableList<Col> getTestCols() {
                 .build())
         .add(
             Col.builder("blob_col", "BLOB", new SourceColumnType("BLOB", new Long[] {10L}, null))
-                .withValue(new byte[] {0x65, 0x66}, new char[] {'6', '5', '6', '6'})
+                .withValue(new byte[] {0x65, 0x66}, "6566")
```
[the remark this answers is not in the document]
Comment (VardhanThigle) [reply v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapperTest.java:253]: Thanks.
I have added a null case for all the types.
A few other edge cases are covered in the exception test cases.

We are not checking with a real DB in the UT, so it might not be possible to test with the limits of blob length, would cover that in integration tests.

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/MysqlJdbcValueMappings.java [hunk 2]
```diff
@@ -71,7 +71,7 @@ public class MysqlJdbcValueMappings implements JdbcValueMappingsProvider {
 
   /* Hex Encoded string for blob types. */
   private static final ResultSetValueMapper<Blob> blobToHexString =
-      (value, schema) -> Hex.encodeHex(value.getBytes(1, (int) value.length()));
```
Comment (Deep1998) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/MysqlJdbcValueMappings.java:74]: We might have to do this for binary and varbinary types also?

File: v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapperTest.java [hunk 3]
```diff
@@ -323,7 +333,7 @@ private static ImmutableList<Col> getTestCols() {
                     "medium_blob_col",
                     "BLOB",
                     new SourceColumnType("MEDIUMBLOB", new Long[] {10L}, null))
-                .withValue(new byte[] {0x65, 0x66}, new char[] {'6', '5', '6', '6'})
+                .withValue(new byte[] {0x65, 0x66}, "6566")
                 .build())
```
Comment (Deep1998) [inline v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapperTest.java:337]: Similarly UTs for binary, varbinay? And maybe YEAR too to catch this in UTs early on?

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/MysqlJdbcValueMappings.java (hunk 2 above)
Comment (VardhanThigle) [reply v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/MysqlJdbcValueMappings.java:74]: Thanks for the catch! Done.

File: v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapperTest.java (hunk 3 above)
Comment (VardhanThigle) [reply v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapperTest.java:338]: We have a UT for bianry and varbinary, I have updated the same. The UT could not catch the issue earlier since we use derby and not actual db for testing.
For `year` we dont have a corresponding type in derby but we have an existing UT that maps it to dummy derby type (this wont be as effective as an actual DB run).

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM
